src/schoology-extractor/tests/data_generation/__main__.py

```diff
@@ -23,6 +37,79 @@
 )
 logger = logging.getLogger(__name__)
 
-NUMBER_OF_USERS = 60
+request_client: RequestClient = RequestClient(
+    arguments.client_key, arguments.client_secret
+)
+
+
+grading_periods = []
+try:
+    grading_periods = get_gradingperiods(request_client)
+except Exception as ex:
+    traceback.print_exception(type(ex), ex, ex.__traceback__)
```

Comment:
At one point I did directly use the `traceback` module, before I realized that `logger.exception` actually captures the stack trace. Here's a simple script that will demonstrate (I had to double-check before writing this comment):

```python
import logging
from os import error
import sys

def this_fails(denominator):
    return 1/denominator

logger = logging.getLogger(__name__)

logging.basicConfig(
    handlers=[logging.StreamHandler(sys.stdout),],
    format="%(asctime)s - %(levelname)s - %(name)s - %(message)s",
    level="INFO",
)

try:
    this_fails(0)
except BaseException as e:
    logger.exception(e)

logger.info("done with demonstration")
```

Reply:
Cool, I'll update